src/datanode/src/instance.rs

```diff
@@ -155,8 +162,12 @@ impl Instance {
             }
 
             Mode::Distributed => {
-                let catalog = Arc::new(catalog::remote::RemoteCatalogManager::new(
+                let engine_manager = Arc::new(MemoryTableEngineManager::new(
```

Comment:
I think `engine_manager` can be lift out of this `match`. `MemoryCatalogManager` can have TableEngineManager as well.